Source/CarthageKit/Project.swift

```diff
@@ -1103,6 +1133,64 @@ public final class Project { // swiftlint:disable:this type_body_length
 					}
 			}
 	}
+
+	/// Returns an array of CompatibilityInfo representing the pinned version
+	/// of each dependency in Cartfile.resolved which is incompatible with
+	/// one or more of the versions specified in a transitive dependency's
+	/// Cartfile.  The incompatible versions are stored in the
+	/// dependencyVersions property of the CompatibilityInfo object.
+	public func verify(resolvedCartfile: ResolvedCartfile? = nil) -> SignalProducer<(), CarthageError> {
+		let resolvedProducer = (resolvedCartfile != nil) ? SignalProducer(value: resolvedCartfile!) : loadResolvedCartfile()
```

Comment:
I'm not so keen on this loading the Cartfile.resolved by itself. I would not expect this from the signature.

Reply:
I agree. I think we should always pass in the resolved cartfile instead of loading it here.